app/javascript/src/components/Editor.tsx

```diff
@@ -118,6 +122,40 @@ class Editor extends React.Component<EditorProps, EditorState> {
     this.model = model
   }
 
+  private handleSceneName = (
+    currentState: any,
+    newStory: any,
+    ) => {
+    let currentNodes = [...currentState.story.nodes]
+    let updatedName: string = ''
+    let latestLink = Object.assign(new DefaultLinkModel(), currentState.story.links[currentState.story.links.length - 1])
+    let sourcePort = Object.assign(new DefaultPortModel(false, ''))
+    let targetNode = Object.assign(new DefaultNodeModel())
```

Comment:
some of these `Object.assign`s strike me as odd. Is this to appease the typescript checker?

Reply:
> some of these `Object.assign`s strike me as odd. Is this to appease the typescript checker?

Sort of. I ran into an issue when trying to access properties when defining these particular types. After some research, I came across [this](https://medium.com/@aems/one-mans-struggle-with-typescript-class-serialization-478d4bbb5826) article. 

It seemed the problem was this: round-trip serialize/deserialize a TypeScript class to JSON, what you get back is not an instance of that type. After using the method this person used, it seemed the types were happy and I was able to access the properties I needed.